### ISSUE TITLE
Please add support for Streams

### ISSUE THREAD

**Issue by jeremywaguet** [bytefish/TinyCsvParser#40]
On my own project I added this class to make it work with Streams to CsvParserExtensions.cs. Please add support for streams :)

```c#
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using TinyCsvParser;
using TinyCsvParser.Mapping;
using TinyCsvParser.Model;

namespace TinyCsvParser
{
    static class StreamCsvParserExtensions
    {
        private static IEnumerable<string> ReadLines(Stream stream, Encoding encoding)
        {
            using (var reader = new StreamReader(stream, encoding, false, 4096, true)) // This memory stream is volontary kept open allow other uses of it
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    yield return line;
                }
            }
        }

        public static ParallelQuery<CsvMappingResult<TEntity>> ReadFromStream<TEntity>(this CsvParser<TEntity> csvParser, Stream stream, Encoding encoding)
            where TEntity : class, new()
        {
            if (stream == null)
            {
                throw new ArgumentNullException("stream");
            }

            var lines = ReadLines(stream, encoding)
                .Select((line, index) => new Row(index, line));

            return csvParser.Parse(lines);
        }
    }
}

```

**Comment by bytefish** [comment]
Please make a Pull Request and I will be happy to merge! 🌞

**Comment by bytefish** [comment]
@jeremywaguet I modified your code a little, so you can pass the buffer size and so on into the method. I decided to use the .NET default ``StreamReader`` configuration to not confuse users. You can easily adjust the settings from the outside, if you need a different behavior!

I have released 2.3.0 with ``ParseFromStream`` method. 

Please reopen the ticket, if there are any problems. 

Thanks for your support!
